capture/src/server.rs

```diff
@@ -48,24 +48,30 @@ where
             .register("rdkafka".to_string(), Duration::seconds(30))
             .await;
 
-        let partition = OverflowLimiter::new(
-            config.overflow_per_second_limit,
-            config.overflow_burst_limit,
-            config.overflow_forced_keys,
-        );
-        if config.export_prometheus {
-            let partition = partition.clone();
-            tokio::spawn(async move {
-                partition.report_metrics().await;
-            });
-        }
-        {
-            // Ensure that the rate limiter state does not grow unbounded
-            let partition = partition.clone();
-            tokio::spawn(async move {
-                partition.clean_state().await;
-            });
-        }
+        let partition = match config.overflow_enabled {
+            false => None,
+            true => {
+                let partition = OverflowLimiter::new(
+                    config.overflow_per_second_limit,
+                    config.overflow_burst_limit,
+                    config.overflow_forced_keys,
+                );
+                if config.export_prometheus {
+                    let partition = partition.clone();
+                    tokio::spawn(async move {
+                        partition.report_metrics().await;
+                    });
+                }
+                {
+                    // Ensure that the rate limiter state does not grow unbounded
+                    let partition = partition.clone();
+                    tokio::spawn(async move {
+                        partition.clean_state().await;
+                    });
```

Comment:
Huh, wondering if we should `Arc` also the `OverflowLimiter`'s `forced_keys` `HashSet`, since that's being cloned here it could be expensive if it's too large.

Reply:
Anyways, seems like that was there from before, not a blocker on this PR.

Reply:
Agreed. `forced_keys` is empty though, supported as a last resort if the volume is so big that plugin-server cannot forward to overflow fast enough. Cloning it as long as it has a zero or a couple of entries might be cheaper than the Arc.
Should probably be superseeded by a redis zset instead.